### ISSUE TITLE
Error: second time provided was later than self at launch

### ISSUE THREAD

**Issue by blazerzar** [kpcyrd/spotify-launcher#37]
I have a problem with `spotify-launcher` not starting at all. It worked yesterday. Today I booted the computer, Spotify opens by default at launch, but it was not working. So I closed it and tried reopening it, but nothing was happening.

After launching `spotify-launcher` from terminal, I got

```bash
[2023-11-21T20:43:42Z INFO  spotify_launcher::config] Loading configuration file at "/etc/spotify-launcher.conf"
Error: second time provided was later than self
```

I tried creating a new config file, but to no avail. Even if I delete, error persists. I tried running with `sudo`. This time there is no error, Spotify GUI opens, but it is just a gray screen.

Also did a full system update, but Spotify still does not work.

System info:
- Distro: EndeavourOS
- Kernel: 6.6.2-arch1-1
- Spotify Launcher: 0.5.1-2

**Comment by kpcyrd** [comment]
I suspect there have been temporary issues with the system time, causing a datetime in the future to be written into `~/.local/share/spotify-launcher/state.json`, which wasn't considered at this line:

https://github.com/kpcyrd/spotify-launcher/blob/f6f914eb4d3f4e89d9d6c8d8db68f708af38ec12/src/main.rs#L28

As a temporary workaround, it should fix itself after running `spotify-launcher --force-update` because this updates the update check timer to your current time again.

**Comment by blazerzar** [comment]
Thanks for quick reply. I tried your suggestion, if the time in that JSON is incorrect, and it works. My computer might have shutdown unexpectedly yesterday.

App also started working after uninstalling Spotify launcher, installing Spotify from AUR, uninstalling it and installing launcher back. Reinstalling the launcher did not work for some reason.

**Comment by kpcyrd** [comment]
Thanks for reporting this, I'm going to fix this later so this edge-case is automatically handled correctly in the future.

Uninstalling and reinstalling the package only replaces the program in /usr/bin/, but to "start from scratch" you need to delete the `~/.local/share/spotify-launcher/` directory in your home folder with all its content.

**Comment by kpcyrd** [comment]
Fixed in 0.5.2
